packages/eslint-plugin/src/rules/shorthand-property-sorting/index.ts

```diff
@@ -0,0 +1,124 @@
+            let importedFromCorrectPackage = null;
+            const program = findProgramRoot(node);
+
+            // make sure node is imported from the correct package
+            const logicCssAndCssMap =
+              node.parent.type === 'CallExpression' &&
+              node.parent.callee.type === 'Identifier' &&
+              (node.parent.callee.name === 'css' || node.parent.callee.name === 'cssMap');
+
+            const logicStyled =
+              node.parent.type === 'CallExpression' &&
+              node.parent.callee.type === 'MemberExpression' &&
+              node.parent.callee.object.type === 'Identifier' &&
+              node.parent.callee.object.name === 'styled';
+
+            const logicXcc =
+              node.parent.type === 'CallExpression' &&
+              node.parent.callee.type === 'Identifier' &&
+              node.parent.callee.name === 'xcss';
+
+            if (program && (logicCssAndCssMap || logicStyled)) {
+              importedFromCorrectPackage = program.body.find(
+                (n) => n.type === 'ImportDeclaration' && n.source.value === '@compiled/react'
+              );
+            }
+            if (program && logicXcc) {
+              importedFromCorrectPackage = program.body.find(
+                (n) => n.type === 'ImportDeclaration' && n.source.value === '@atlaskit/primitives'
+              );
+            }
+
+            // loop through the css properties of a ObjectExpression object
+            node.properties.some((property) => {
+              if (property.type === 'Property' && property.key.type === 'Identifier') {
+                const depth = shorthandFor[property.key.name];
+
+                // if we find a property with a with a higher depth below one with a lower depth, we trigger the eslint error
+                if (depth < lowestDepth) {
+                  fixRequired = true;
+                  return;
+                } else {
+                  lowestDepth = depth;
+                }
+              }
+            });
+
+            if (fixRequired && importedFromCorrectPackage) {
+              context.report({
+                node: node,
+                messageId: 'shorthand-first',
+                fix: (fixer) => {
+                  // sort the properties by depth
+                  const sortedProperties = node.properties.slice().sort((a, b) => {
+                    if (
+                      a.type === 'Property' &&
+                      a.key.type === 'Identifier' &&
+                      b.type === 'Property' &&
+                      b.key.type === 'Identifier'
+                    ) {
+                      return shorthandFor[a.key.name] - shorthandFor[b.key.name];
+                    }
+                    return 0;
+                  });
+
+                  const sourceCode = context.getSourceCode();
+                  const sortedCode = sortedProperties
+                    .map((property) => sourceCode.getText(property))
```

Comment:
can we have a test that includes pseudo-selectors (e.g. `&:hover`), pseudo-elements (e.g. `&::after`), and media queries? e.g.

```tsx
const styles = css({
    '&:hover': {
        color: 'blue',
        fontWeight: 'bold',
        font: '...',
    },
    fontWeight: 'bold',
    font: '...',
    color: 'blue',

})
```

note that eslint does not apply overlapping fixes, so you'll get wonky test results if you try to sort the stuff inside `&:hover` and the stuff outside `&:hover` at the same time (that's ok - outside of tests, eslint will re-run the eslint rule over and over until the rule no longer suggests a fix)